happi/cli.py

```diff
@@ -102,13 +105,58 @@ def happi_cli(args):
                 return
             if value.replace('.', '').isnumeric():
                 logger.debug('Changed %s to float', value)
-                value = float(value)
-            client_args[criteria] = value
+                value = str(float(value))
 
-        results = client.search(**client_args)
-        if results:
+            if criteria == 'z' and ',' in value:
+                start = None
+                end = None
+                z_is_range = True
+                try:
+                    start, end = value.replace(
+                        '[', '').replace(']', '').split(',')
+                    start = float(start)
+                    end = float(end)
+                except Exception as ex:
+                    logger.error("Invalid numbers for the z range %s", ex)
+                    sys.exit(1)
+                if start < end:
+                    results += client.search_range('z', start, end)
+                else:
+                    logger.error('Invalid z range')
+
+            # skip the criteria for range values
+            # it won't be a valid criteria for search_regex()
+            if z_is_range and criteria == 'z':
+                pass
+            else:
+                client_args[criteria] = fnmatch.translate(value)
```

Comment:
I think sticking to the unix glob for the cli was a good call

Reply:
Yeah thanks for this idea, i think you mentioned it in the Issue description and @klauer also elaborated on it - thank you, i like this too